pkg/ratelimiter/ratelimiterset.go

```diff
@@ -0,0 +1,46 @@
+package ratelimiter
+
+import (
+	"github.com/bookingcom/nanotube/pkg/conf"
+	"github.com/bookingcom/nanotube/pkg/metrics"
+	"sync"
+	"time"
+)
+
+// Set consists of all the RateLimiters in use.
+type Set struct {
+	containerLock         sync.RWMutex
+	containerRateLimiters map[string]RateLimiter
+	globalRateLimiter     RateLimiter
+}
+
+// NewRateLimiterSet creates a new ratelimiter Set using main configuration.
+func NewRateLimiterSet(cfg *conf.Main, prom *metrics.Prom) *Set {
+	rls := &Set{
+		containerRateLimiters: make(map[string]RateLimiter),
+	}
+	if cfg.RateLimiterGlobalRecordLimit > 0 {
+		windowSize := time.Duration(cfg.RateLimiterWindowSizeSec) * time.Second
+		rls.globalRateLimiter = NewSlidingWindowRateLimiter(windowSize, int64(cfg.RateLimiterGlobalRecordLimit), prom.GlobalRateLimiterBlockedRecords)
+	}
+	return rls
+}
+
+// GetOrCreateContainerRateLimiterWithID creates a container RateLimiter by id and main configuration
+// if it doesn't exist. It'll return the RateLimiter for the corresponding container.
+func (s *Set) GetOrCreateContainerRateLimiterWithID(id string, cfg *conf.Main, prom *metrics.Prom) RateLimiter {
+	s.containerLock.Lock()
```

Comment:
It looks like this lock may not be necessary. The limiters of all containers are independent. The lock would be needed to sync the map, but there's also no reason to keep the map — the limiters for individual containers can be passed down to the data reading functions.